lib/internal/Magento/Framework/View/Page/Config.php

```diff
@@ -463,7 +463,7 @@ public function addRss($title, $href)
      */
     public function addBodyClass($className)
     {
-        $className = preg_replace('#[^a-z0-9]+#', '-', strtolower($className));
+        $className = preg_replace('#[^a-z0-9-]+#', '-', strtolower($className));
```

Comment:
Maybe add _ to the regex after - for classes like class--1?

Reply:
Hi Miguel,
done!